app/models/product_specific_rule.rb

```diff
@@ -0,0 +1,38 @@
+require 'api_entity'
+
+class ProductSpecificRule
```

Comment:
```suggestion
class DeclarableSpecificRule
```

Product isn't really a concept/naming we have currently